src/GitTfs/Core/Ext.cs

```diff
@@ -211,17 +211,24 @@ public static IEnumerable<T[]> ToBatch<T>(this IEnumerable<T> source, int batchS
         /// </summary>
         /// <param name="source">The source sequence.</param>
         /// <param name="action">The action to execute.</param>
+        /// <param name="noParallel">set to true> to disable parallel processing</param>
         /// <typeparam name="T">The source item type</typeparam>
-        public static void DoParallel<T>(this IEnumerable<T> source, Action<T> action)
```

Comment:
I'm not very fond of having a method named DoParallel to do something sequentially.

Is it possible to either find a more meaningful name  or not call this method but a new one doing the action sequentially?

Reply:
Agreed.

I would like rename it  to `ForEach` - since essentially it does mostly the same as [`List<T>.ForEach(Action<T>)`](https://docs.microsoft.com/de-de/dotnet/api/system.collections.generic.list-1.foreach)

Then we could split in `ForEach` and `ParallelForEach` or have an additional parameter (I'd reverse the meaning for that case..) something along the lines of `ForEach(Action<T>, bool parallelizeActions)`

Reply:
That seems good to me.

Reply:
done in 55da262